### PR TITLE
sqlserver: reuse running wal-g proxy service

### PR DESCRIPTION
**Pull request by mialinx**
###  SQLServer

Option to reuse wal-g proxy running as system service

This will allow run several wal-g command (like backup-push and log-push) in parallel.

### CODE REVIEW

File: internal/databases/sqlserver/sqlserver.go [hunk 1]
```diff
@@ -493,3 +497,38 @@ func ListBackupProperties(db *sql.DB, urls string, logBackupName string) ([]*Bac
...
+		tracelog.ErrorLogger.PrintOnError(lw.c.Close())
+	}
+}
+
+func RunOrReuseProxy(ctx context.Context, cancel context.CancelFunc, folder storage.Folder) (*LockWrapper, error) {
+	bs, err := blob.NewServer(folder)
+	if err != nil {
+		return nil, xerrors.Errorf("proxy create error: %v", err)
+	}
+	reuse, _, err := internal.GetBoolSetting(internal.SQLServerUseProxyService)
```
Comment (usernamedt) [inline internal/databases/sqlserver/sqlserver.go:516]: We usually retrieve the config settings at the top level.

File: docs/SQLServer.md [hunk 2]
```diff
@@ -122,3 +122,12 @@ wal-g delete retain 3
...
 wal-g delete everything
 ```
+
+Proxy as Service
+-----------------
+By default any wal-g command, like backup-push, runs proxy in background for the duration of the command.
+Alternatively wal-g commands may use proxy started as system service.
+In order to do this you need to start `wal-g proxy` as system service (or daemon on linux) and add following config options:
+```bash
+SQLSERVER_USE_PROXY_SERVICE: True
```
Comment (usernamedt) [inline docs/SQLServer.md:132]: I see that in code it's actually called 'reuse'

**Review by usernamedt — APPROVED** [review]
seems OK

File: docs/SQLServer.md (hunk 2 above)
Comment (mialinx) [reply docs/SQLServer.md]: Name fixed

**Pull request merged**